### PR TITLE
[UPG] Working on version 8

### PR DESCRIPTION
**Pull request by gfcapalbo**
Migrated to version 8, tested working modules:

stock_delivery_times_advanced                [MODIFIED]
stock_delivery_times_delivery_date_on_move        [NO MODIFICATIONS]
stock_delivery_times_reschedule               [NO MODIFICATIONS]
stock_delivery_times_working_days    [MODIFIED]

### CODE REVIEW

File: stock_delivery_times_advanced/product.py [hunk 1]
```diff
@@ -82,13 +82,24 @@ def _get_delays(self, cr, uid, product, qty=1, context=None):
...
+            supplierinfo_obj = self.pool.get('product.supplierinfo')
+            delay = (product.product_tmpl_id.seller_delay or 0.0) + product.product_tmpl_id.sale_delay
+            all_supplierinfos = product.product_tmpl_id.seller_ids
+            mainseller = product.product_tmpl_id.seller_id
+            if all_supplierinfos:
+                if mainseller:
+                    for supplierinfo in all_supplierinfos:
+                        if supplierinfo.name == mainseller:
+                            mainsupplierinfo = supplierinfo
+                    mainsupplierinfo_obj = supplierinfo_obj.browse(cr, uid, mainsupplierinfo.id)
```
Comment (hbrunn) [inline stock_delivery_times_advanced/product.py:97]: this goes wrong if `mainseller not in all_supplierinfos`. And you shouldn't rebrowse browse records you already have.

Say `mainsupplierinfo = None` before the loop and remove this line

File: stock_delivery_times_advanced/product.py [hunk 2]
```diff
@@ -82,13 +82,24 @@ def _get_delays(self, cr, uid, product, qty=1, context=None):
...
+                else:
```
Comment (hbrunn) [inline stock_delivery_times_advanced/product.py:98]: replace by `if not mainsupplierinfo:`

File: stock_delivery_times_advanced/product.py [hunk 3]
```diff
@@ -82,13 +82,24 @@ def _get_delays(self, cr, uid, product, qty=1, context=None):
...
+                    mainsupplierinfo_obj = all_supplierinfos[0]
```
Comment (hbrunn) [inline stock_delivery_times_advanced/product.py:99]: `mainsupplierinfo = all_supplierinfos and all_supplierinfos[0] or None` or raise an error if there are no supplierinfos

File: stock_delivery_times_advanced/product.py [hunk 4]
```diff
@@ -82,13 +82,24 @@ def _get_delays(self, cr, uid, product, qty=1, context=None):
...
+                if mainsupplierinfo.supplier_shortage:
+                    # TODO use a different calendar for the supplier delay than the company calendar
+                    supplier_shortage = mainsupplierinfo_obj['supplier_shortage']
```
Comment (hbrunn) [inline stock_delivery_times_advanced/product.py:102]: mainsupplierinfo.supplier_shortage
(you can use both syntaxes on browse_records, but the member syntax is way more common)

File: stock_delivery_times_working_days/sale.py [hunk 5]
```diff
@@ -40,6 +40,7 @@ def _get_date_planned(self, cr, uid, order, line, start_date, context=None):
         """
         cal_obj = self.pool['resource.calendar']
         start_date = self._get_start_date(cr, uid, order, line, start_date, context=context)
+        start_date = datetime.strptime(start_date[:10], DEFAULT_SERVER_DATE_FORMAT)
```
Comment (hbrunn) [inline stock_delivery_times_working_days/sale.py:43]: how can this work? `self._get_start_date` seems to return a datetime?

**Comment by bealdav** [comment]
@gfcapalbo @hbrunn, nice to see you working on v8 on this matter. On my side I'm working on 'mrp capacity' in v7. I need this method
https://github.com/gfcapalbo/openerp-stock-delivery-times/blob/3848e8875667228f38aca06c11f76462111e8d09/stock_delivery_times_working_days/resource.py
It isn't concern only the stock, it's more general purpose. 
Could we consider to extract it in another module like 'resource_helper' or 'calendar_helper' ?
If your are ok on the name, I can propose on OCA a v7 module. In which repo ? server-tool ? What do you think ?
copy to @bguillot
Thanks for your feedback

**Comment by hbrunn** [comment]
`resource_*` would be good because the base module is resource. But I'm not sure if one function is worth its own module, or is it the plan to add more convenience functions there?

**Comment by bealdav** [comment]
@hbrunn,  I understand, it is short for a module, but how can I reuse everywhere we need to deal with date (working date), if I don't put it in a module ? You're right it could be a module which add more convenient methods about resource in the future. 
like this one in an other context
https://github.com/OCA/partner-contact/blob/8.0/partner_helper/__openerp__.py

**Pull request closed without merging**